examples/saving_collection.py

```diff
@@ -0,0 +1,17 @@
+import os
+import sys
+
+sys.path.insert(0, os.getenv("BUILDTEST_ROOT"))
+
+from buildtest.modules.util import Module
+
+a = Module(["zlib"])
```

Comment:
Why is this required to be a list? Intuitively, I'd just provide a string.

Reply:
My thought was around dealing with module load command

```
self.module_load_cmd = [f"module load {x} && " for x in self.modules]
```

If user specify 
```
a = Module(["zlib","GCCcore","CUDA/10.0"]
```

This should generate a module load command as follows:

```
module load zlib && module load GCCcore && module load CUDA/10.0
```

It was easier to process a list since i need to run module load against all modules. Note that running ``module load zlib GCCcore CUDA/10.0`` is also valid but it does not guarantee all the modules are loaded. It is important when it comes to checking return code of the module load command.

You can try this in your system and see output of ``echo $?``  with command chaining like how i propose version doing ``module load <a> <b> <c>``.

Initialliy i had done ``module load <a>; module load <b>; module load <c>`` and this didnt work because it did not guarantee all modules were loaded since exit code will be of the last command.

Also in order for me to implement the ``Module().save()`` method i needed modules to be loaded in order to save them.

So running ``module load zlib && module save zlib`` worked when passing this to ``BuildTestCommand``

Reply:
Sorry, what I mean is to allow:

```python
a = Module("zlib")
```
And then in the init, do something like:

```python
if not isinstance(name, list):
    name = [name]
```
That way, both of these will work:

```python
a = Module("zlib")
b = Module(["zlib", "mpi"])
```
Because it's not super intuitive to provide a list with one item to a function instantiation. The internal stuffs that you detailed looks good to me :)

Reply:
got it. Let me fix this i think i get what you mean.